### PR TITLE
CI: Automate colima delete command with confirmation

### PR DESCRIPTION
**Pull request by SmartManoj**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**

image

[Action link](https://github.com/OpenDevin/OpenDevin/actions/runs/10041954833/job/27751219274)

---

**Give a summary of what the PR does, explaining any non-trivial design decisions**

Increased ATTEMPT_LIMIT to 4 to check the inference ( QEMU works in 3rd attempt)  https://github.com/lima-vm/lima/pull/2455#issuecomment-2236128317

---
**Other references**

### CODE REVIEW

**Review by neubig — COMMENTED** [review]
Thanks for attempting to fix the Colima problems! But sorry, could you explain more?

Why do you think that 3 attempts may be insufficient but 4 would be better?

Also, why delete after 3 attempts when the maximum number of attempts is 4?

**Comment by SmartManoj** [comment]
> Increased ATTEMPT_LIMIT to 4 to check the inference ( QEMU works in 3rd attempt) [lima-vm/lima#2455 (comment)](https://github.com/lima-vm/lima/pull/2455#issuecomment-2236128317)

first two attempts failed with msg="[hostagent] QEMU did not exit in 3m0s, forcibly killing QEMU"
third one with msg="error starting docker: cannot restart, VM not previously started" (inference => 1st essential requirement is satisfied in ~7mins (3+3+1))

> Why do you think that 3 attempts may be insufficient but 4 would be better?

The timeout issue persists for 2 attempts. So, need to check `colima stop` or `colima delete` is efficient.

So on 3rd attempt, it will stop, and 4th attempt it will delete.

> Also, why delete after 3 attempts when the maximum number of attempts is 4?

a fall back method

**Comment by SmartManoj** [comment]
[Another failed action](https://github.com/OpenDevin/OpenDevin/actions/runs/10075489341/job/27853871846)

**Comment by neubig** [comment]
Thanks! But do we know why `colima` is failing? Do we have any evidence that this actually fixes the problem? It seems like this is trying things out without any guarantee that this actually fixes the problem...

**Comment by SmartManoj** [comment]
> do we know why `colima` is failing?

Due to Github actions resource contention? https://github.com/lima-vm/lima/pull/2455#discussion_r1666594901

**Comment by neubig** [comment]
I think this is probably fixed by #3097 , so let's close this for now. If we still end up getting flaky tests we could reconsider!

**Comment by SmartManoj** [comment]
> I think this is probably fixed by #3097

It didn't. https://github.com/OpenDevin/OpenDevin/pull/3097#issuecomment-2249859902

File: .github/workflows/run-unit-tests.yml [hunk 1]
```diff
@@ -99,8 +99,8 @@ jobs:
               break
             else
               echo "Failed to start Colima. Attempt $i/$ATTEMPT_LIMIT."
-              if [ $i -eq $ATTEMPT_LIMIT ]; then
-                colima delete
+              if [ $i -eq $(($ATTEMPT_LIMIT - 1)) ]; then
```
Comment (SmartManoj) [inline .github/workflows/run-unit-tests.yml:102]: CC: @tobitege

Comment (tobitege) [reply .github/workflows/run-unit-tests.yml:102]: Shouldn't it delete after every failed attempt (except the last one), like this?

```py
if [ $i -lt $ATTEMPT_LIMIT ]; then
```

Comment (SmartManoj) [reply .github/workflows/run-unit-tests.yml:102]: Yes.

Comment (tobitege) [reply .github/workflows/run-unit-tests.yml:102]: Ok, I'll change that in my PR then

**Pull request closed without merging**